test/lib/pmd/PmdEngine.test.ts

```diff
@@ -10,11 +11,18 @@ import * as TestOverrides from '../../test-related-lib/TestOverrides';
 import { CUSTOM_CONFIG } from '../../../src/Constants';
 import * as DataGenerator from '../eslint/EslintTestDataGenerator';
 
+Messages.importMessagesDirectory(__dirname);
+const engineMessages = Messages.loadMessages('@salesforce/sfdx-scanner', 'PmdEngine');
+
 TestOverrides.initializeTestSetup();
 class TestPmdEngine extends PmdEngine {
 	public processStdOut(stdout: string): RuleResult[] {
 		return super.processStdOut(stdout);
 	}
+
+	public processStdErr(stderr: string): string {
```

Comment:
I've tried out accessing private members using the `as any` trick to avoid having create a new subclass. This allows the members to be private instead of protected. This is an example https://github.com/forcedotcom/sfdx-scanner/blob/dev/test/lib/RuleFilter.test.ts#L40 I found it useful in case you want to try it out.

Reply:
Interesting idea. I'll see if it works with private methods.